adaptation.lua

```diff
@@ -0,0 +1,128 @@
+-- Original opening comment before lua adaptation
+-- View or set cavern adaptation levels
+-- based on removebadthoughts.rb
+
+-- Rewritten by TBSTeun using OpenAI GPT from adaptation.rb
+
+local help = [====[
+
+adaptation
+==========
+View or set level of cavern adaptation for the selected unit or the whole fort.
+
+Usage::
+
+    adaptation <show|set> <him|all> [value]
+
+The ``value`` must be between 0 and 800,000 (inclusive).
+
+]====]
+
+-- Color constants, values mapped to color_value enum in include/ColorText.h
+local COLOR_RESET = -1
+local COLOR_GREEN = 2
+local COLOR_RED = 4
+local COLOR_YELLOW = 14
+
+local function print_color(color, s)
+    dfhack.color(color)
+    dfhack.print(s)
+    dfhack.color(COLOR_RESET)
+end
+
+local function usage(s)
+    if s ~= nil then
+        print(s)
+    end
+    print("Usage: adaptation <show|set> <him|all> [value]")
+end
+
+local args = {...}
+
+local mode = args[1] or "help"
+local who = args[2]
+local value = args[3]
+
+if mode == "help" then
+    usage()
+elseif mode ~= "show" and mode ~= "set" then
+    usage("Invalid mode '" .. mode .. "': must be either 'show' or 'set'")
+end
+
+if who == nil then
+    usage("Target not specified")
+elseif who ~= "him" and who ~= "all" then
+    usage("Invalid target '" .. who .. "'")
+end
+
+if mode == "set" then
+    if value == nil then
+        usage("Value not specified")
+    elseif not tonumber(value) then
+        usage("Invalid value '" .. value .. "'")
+    end
+
+    if tonumber(value) < 0 or tonumber(value) > 800000 then
+        usage("Value must be between 0 and 800000")
+    end
+    value = tonumber(value)
+
+end
+
+local num_set = 0
+
+local function set_adaptation_value(unit, v)
```

Comment:
Try to put functions above executing code

Reply:
done